plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java

```diff
@@ -884,7 +884,7 @@ public int compare(Viewer viewer, Object e1, Object e2) {
             if (e1 instanceof DBNDatabaseNode n1 && e2 instanceof DBNDatabaseNode n2) {
                 Object o1 = n1.getValueObject();
                 Object o2 = n2.getValueObject();
-                if (o1 instanceof DBPObjectWithOrdinalPosition c1 && o2 instanceof DBPObjectWithOrdinalPosition c2) {
+                if (o1 instanceof DBSTableIndexColumn c1 && o2 instanceof DBSTableIndexColumn c2) {
```

Comment:
I think DBSEntityAttributeRef + DBPObjectWithOrdinalPosition is better.
Anyhow it is a bad idea to add this logic in the default comparator. User may forcibly reorder index columns by any other attribute.

We order table columns correctly (by ordinal position, as they come from database model). 
How index columns are different?

Also let's add index column index in the list of visible columns (currently we don't show it at all)